db/skipped_sequence.go

```diff
@@ -42,20 +44,37 @@ type SkippedSequenceSlice struct {
 }
 
 var _ SkippedSequenceListEntry = &SingleSkippedSequence{}
+var _ SkippedSequenceListEntry = &SkippedSequenceRange{}
```

Comment:
I agree with this idea, and am fine doing it in a later optimization.  However, it does make me wonder whether anything that's a no-op belongs on the interface.  (setStartSeq/setLastSeq/extendRange).  Looking at the code, this goes back to the fact that we're using isRange instead of a type switch.  Was this intentional for performance reasons?